src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests.cs

```diff
@@ -728,6 +728,691 @@ public object WProp
+";
+            var comp = CreateStandardCompilation(source, parseOptions: TestOptions.Regular7_2);
+            comp.VerifyDiagnostics(
+                // (7,5): warning CS8360: Field-targeted attributes on auto-properties are not supported in language version 7.2. Please use language version 7.3 or greater.
+                //     [field: System.Obsolete]
+                Diagnostic(ErrorCode.WRN_AttributesOnBackingFieldsNotAvailable, "[field: System.Obsolete]").WithArguments("7.2", "7.3").WithLocation(7, 5),
+                // (8,5): warning CS8360: Field-targeted attributes on auto-properties are not supported in language version 7.2. Please use language version 7.3 or greater.
+                //     [field: A]
+                Diagnostic(ErrorCode.WRN_AttributesOnBackingFieldsNotAvailable, "[field: A]").WithArguments("7.2", "7.3").WithLocation(8, 5),
+                // (11,5): warning CS8360: Field-targeted attributes on auto-properties are not supported in language version 7.2. Please use language version 7.3 or greater.
+                //     [field: System.Obsolete("obsolete", error: true)]
+                Diagnostic(ErrorCode.WRN_AttributesOnBackingFieldsNotAvailable, @"[field: System.Obsolete(""obsolete"", error: true)]").WithArguments("7.2", "7.3").WithLocation(11, 5)
+                );
+        }
+
+        [Fact]
+        public void TestFieldAttributesOnAutoProperty()
+        {
+            string source = @"
+[System.AttributeUsage(System.AttributeTargets.Field, AllowMultiple = true) ]
+public class A : System.Attribute
+{
+    public A(int i) { }
+}
+[System.AttributeUsage(System.AttributeTargets.Property, AllowMultiple = true) ]
+public class B : System.Attribute
+{
+    public B(int i) { }
+}
+
+public class Test
+{
+    [field: A(1)]
+    public int P { get; set; }
+
+    [field: A(2)]
+    public int P2 { get; }
+
+    [B(3)]
+    [field: A(33)]
+    public int P3 { get; }
+
+    [property: B(4)]
+    [field: A(44)]
+    [field: A(444)]
+    public int P4 { get; }
+}
+";
+            Func<bool, Action<ModuleSymbol>> symbolValidator = isFromSource => moduleSymbol =>
+            {
+                var accessorsExpected = isFromSource ? new string[0] : new[] { "System.Runtime.CompilerServices.CompilerGeneratedAttribute" };
+
+                var @class = moduleSymbol.GlobalNamespace.GetMember<NamedTypeSymbol>("Test");
+
+                var prop1 = @class.GetMember<PropertySymbol>("P");
+                Assert.Empty(prop1.GetAttributes());
+                AssertEx.SetEqual(accessorsExpected, GetAttributeStrings(prop1.GetMethod.GetAttributes()));
+                AssertEx.SetEqual(accessorsExpected, GetAttributeStrings(prop1.SetMethod.GetAttributes()));
+
+                if (isFromSource)
+                {
+                    var field1 = @class.GetMember<FieldSymbol>("<P>k__BackingField");
+                    AssertEx.SetEqual(accessorsExpected.Concat(new[] { "A(1)" }), GetAttributeStrings(field1.GetAttributes()));
```

Comment:
>SetEqual [](start = 29, length = 8)

This verification should not be limited to source. There is a way to instruct compilation to import privates.

Reply:
Applies to all similar cases below.

---
In reply to: [148158621](https://github.com/dotnet/roslyn/pull/22664#discussion_r148158621) [](ancestors = 148158621)